katdal/chunkstore_s3.py

```diff
@@ -70,13 +70,14 @@ def __init__(self, client):
         if not botocore:
             raise _botocore_import_error
         error_map = {EndpointConnectionError: StoreUnavailable,
+                     ConnectionError: StoreUnavailable,
                      client.exceptions.NoSuchKey: ChunkNotFound,
                      client.exceptions.NoSuchBucket: ChunkNotFound}
         super(S3ChunkStore, self).__init__(error_map)
         self.client = client
 
     @classmethod
-    def from_url(cls, url, **kwargs):
+    def from_url(cls, url, timeout=5, **kwargs):
```

Comment:
5 is possibly a little low. I don't know about S3, but I seem to recall other places where I've had to increase 5 second connection timeouts to make things reliable.

Reply:
The default has an exponential backoff that takes *ages* to time out. I could be persuaded to make it 10 seconds, but limited testing never needed more than a second to connect.